### PR TITLE
Add is:tentative command

### LINKED ISSUE #36: Feature Request: Disregard or elsehow mark tentative tests.
_From @otherdaniel on August 25, 2017 9:14_

Thanks for the dashboard!

Please add some feature to disregard tentative tests (or maybe account for them separately), so that implementing a proposed feature won't count "against" browsers.

Example: I am currently doing a first draft implementation of extending SRI (subresource integrity) with digital signatures, as proposed to (but not ratified by) the W3C webappsec group. To support that work, I have added webplatform tests for this feature, in the subresource-integrity dir. The dashboard now makes it look like browser are behind on implementing this, despite it merely being a proposed feature.

References:
- Explainer for feature in W3C webappsec:
https://github.com/w3c/webappsec-subresource-integrity/blob/master/signature-based-restrictions-explainer.markdown
- Tentative WPT tests for this feature on the wptdashboard: http://wpt.fyi/subresource-integrity

Also, maybe related: Chrome/Chromium seems to run wpt tests with experimental features enabled, while your dashboard runs the release config. That's probably the better choice; but making both, release + experimental result sets available might be useful.

_Copied from original issue: w3c/wptdashboard#99_

### PR DESCRIPTION
**Pull request by stephenmcgruer**
Fixes #36

### CODE REVIEW

File: api/query/cache/index/filter.go [hunk 1]
```diff
@@ -305,11 +305,28 @@ func (l Link) Filter(t TestID) bool {
...
+		set := mapset.NewSet()
+		for _, result := range q.runResults {
+			set.Add(result.GetResult(t))
+		}
+		return set.Cardinality() > 1
+	case query.MetadataQualityTentative:
+		// is:tentative only returns rows from tests with .tentative.
+		// in their name. See
+		// https://web-platform-tests.org/writing-tests/file-names.html
+		name, subtest, err := q.tests.GetName(t)
```
Comment (Hexcles) [inline api/query/cache/index/filter.go]:
```suggestion
		name, _, err := q.tests.GetName(t)
```

**Comment by stephenmcgruer** [comment]
Ahaha, appreciate the fast review but this was meant to be a draft (whoops). Will re-ping when ready for review.

**Comment by foolip** [comment]
How about doing `is:optional` as well? :)

**Comment by stephenmcgruer** [comment]
Ok PTAL @Hexcles (low priority). It has tests now! :D

**Pull request merged**